types/react/index.d.ts

```diff
@@ -88,7 +88,7 @@ declare namespace React {
     }
 
     interface ReactElement<P> {
-        type: string | ComponentClass<P> | SFC<P>;
+        type: string | symbol | number | ComponentClass<P> | SFC<P>;
```

Comment:
Are you suggesting that I also add a new type alias named `Fragment` that is `symbol | number` and then use that instead as shorthand?

Reply:
@uniqueiniquity sorry did not see your comment
> new type alias named `Fragment` that is `symbol | number`

Yes, you have also `const Fragment: symbol | number;`. It's a suggestion, your choice.

Reply:
I'm a little confused by your suggestion; that's a value with the type `symbol | number`, not a type alias.

Reply:
> that's a value [...] not a type alias

I know, you can then write `const Fragment: NEW_TYPE;`, NEW_TYPE being named something like `ReactFragment` since I guess naming it `Fragment` would clashname with the `const Fragment...`.

FYI a type `Fragment` exist in React Flow source code: https://github.com/facebook/react/blob/v16.1.0/packages/shared/ReactTypes.js#L18

Reply:
Aha, I understand now. I would prefer keeping it explicit here to prevent confusion with [this type](https://github.com/uniqueiniquity/DefinitelyTyped/blob/4d07771b6c5ed5c3b75c7ee3e0d939557927403d/types/react/index.d.ts#L169).